Alignment/OfflineValidation/src/PVValidationHelpers.cc

```diff
@@ -0,0 +1,276 @@
+      break;
+
+    // normalized
+
+    case PVValHelper::norm_dxy  :
+      returnType =  std::make_tuple("norm_dxy","d_{xy}/#sigma_{d_{xy}}","");
+      break;
+    case PVValHelper::norm_dx   :
+      returnType =  std::make_tuple("norm_dx","d_{x}/#sigma_{d_{x}}","");
+      break;
+    case PVValHelper::norm_dy   :
+      returnType =  std::make_tuple("norm_dy","d_{y}/#sigma_{d_{y}}","");
+      break;
+    case PVValHelper::norm_dz   :
+      returnType =  std::make_tuple("norm_dz","d_{z}/#sigma_{d_{z}}","");
+      break;
+    case PVValHelper::norm_IP2D :
+      returnType =  std::make_tuple("norm_IP2D","IP_{2D}/#sigma_{IP_{2D}}","");
+      break;
+    case PVValHelper::norm_resz :
+      returnType =  std::make_tuple("norm_resz","z_{trk}-z_{vtx}/#sigma_{res_{z}}","");
+      break;
+    case PVValHelper::norm_IP3D :
+      returnType =  std::make_tuple("norm_IP3D","IP_{3D}/#sigma_{IP_{3D}}","");
+      break;
+    case PVValHelper::norm_d3D  : 
+      returnType =  std::make_tuple("norm_d3D","d_{3D}/#sigma_{d_{3D}}","");
+      break;
+
+    default:
+      edm::LogWarning("PVValidationHelpers") <<" getTypeString() unknown residual type: "<<type<<std::endl;
+    }
+
+  return returnType;
+  
+}
+
+//*************************************************************
+PVValHelper::plotLabels PVValHelper::getVarString (PVValHelper::plotVariable var)
+//*************************************************************
+{
+  PVValHelper::plotLabels returnVar;
+  switch(var)
+    {
+    case PVValHelper::phi  :
+      returnVar =  std::make_tuple("phi","#phi","[rad]");
+      break;
+    case PVValHelper::eta   :
+      returnVar =  std::make_tuple("eta","#eta","");
+      break;
+    case PVValHelper::pT   :
+      returnVar = std::make_tuple("pT","p_{T}","[GeV]");
+      break;
+    case PVValHelper::pTCentral :
+      returnVar = std::make_tuple("pTCentral","p_{T} |#eta|<1.","[GeV]");
+      break;
+    case PVValHelper::ladder   :
+      returnVar = std::make_tuple("ladder","ladder number","");
+      break;
+    case PVValHelper::modZ :
+      returnVar = std::make_tuple("modZ","module number","");
+      break;
+    default:
+      edm::LogWarning("PVValidationHelpers") <<" getVarString() unknown plot variable: "<<var<<std::endl;
+    }
+
+  return returnVar;
+  
+}
+
+//*************************************************************
+std::vector<float> PVValHelper::generateBins(int n, float start,float range)
+//*************************************************************
+{
+
+  std::vector<float> v(n);
+  float interval = range/(n-1);
+  std::iota(v.begin(),v.end(),1.);
+
+  //std::cout<<" interval:"<<interval<<std::endl;
+  //for(float &a : v) { std::cout<< a << " ";  }
+  //std::cout<< "\n";
+
+  std::for_each(begin(v), end(v), [&](float& a) { a = start+((a-1)*interval); });
+
+  return v;
+
+}
+
+//*************************************************************
+Measurement1D PVValHelper::getMedian(TH1F *histo)
```

Comment:
hi @mmusich - you can use TH1F::GetQuantiles to avoid the memory copies (with a someone awkward interface..